src/JsonTask.php

```diff
@@ -63,9 +66,15 @@ public function jsonSerialize()
     public static function restore($data)
     {
         $taskData = json_decode($data, true);
+
+        if (!isset($taskData['invocable'], $taskData['params'])){
+            return null;
+        }
+
         $task = new self($taskData['invocable'], $taskData['params']);
+
         if (isset($taskData['report'])) {
-            $task->setReport($taskData['report']);
+            $task->setReport(\common_report_Report::jsonUnserialize($taskData['report']));
```

Comment:
This will break existing code which expects to give array as report. For example:
https://github.com/oat-sa/extension-tao-testqti/blob/master/actions/class.RestQtiTests.php#L154